pkg/fanal/analyzer/language/python/packaging/packaging.go

```diff
@@ -65,7 +66,7 @@ func (a packagingAnalyzer) PostAnalyze(_ context.Context, input analyzer.PostAna
 	var apps []types.Application
 
 	required := func(path string, _ fs.DirEntry) bool {
-		return filepath.Base(path) == "METADATA" || isEggFile(path)
+		return filepath.Base(path) == "METADATA" || isEggFile(path) || slices.Contains(input.FilePatterns, path)
```

Comment:
A file may not end with `.egg` if `--file-patterns` is specified. But I don't have a simple idea of how to handle that. We may want to ignore the edge case.
https://github.com/aquasecurity/trivy/blob/28ac1caafa18119d78f4ce0161e5121ce0531dd5/pkg/fanal/analyzer/language/python/packaging/packaging.go#L79

Reply:
We may need to split this analyzer into two analyzers for `.egg` and `.egg-info`.

Reply:
> We may need to split this analyzer into two analyzers for .egg and .egg-info.

I agree with this decision.
Split analyzers in https://github.com/aquasecurity/trivy/pull/7365/commits/afb738737c72774849c3d701da92244d3b0b28ae

Reply:
Can we make it a separate PR?

Reply:
I've developed a nasty habit of adding all the changes to one PR...

I created #7514 and reverted the changes in this PR - https://github.com/aquasecurity/trivy/pull/7365/commits/2b4f23b93a65d00dbc7c5bd919da3f800c04124f
After merging one of these PRs - I will update the second PR.
Tell me which PR you prefer to merge first

Reply:
#7514 looks easier to merge.

Reply:
#7514 has been merged.
I merged main branch and updated `packaging` PostAnalyzer in https://github.com/aquasecurity/trivy/pull/7365/commits/a2cfee5cb233ef603d43db28c3ab32c4430db3c2